first working version of sort paths

sstefanov opened the pull request:
Hello
I added function for laser cutting:
For each subpath is calculated how many other subpaths it contains.
Then subpaths are sorted by this number. This way first are cut paths which contain no other paths, next these containing 1 path, etc.

sstefanov committed:
first working version of sort paths

paulusjacobus commented:
Hi Stefanov,

Thanks for making this available but I do get one error: from svgpathtools import * 
the svgpathtools is not defined and the scripts abort from that. Please can you update the code?
Thanks heaps!

sstefanov committed:
added install svgpathtools

sstefanov commented:
Hello, Paulus

Sorry, I forgot to mention that you must install svgpathtools from here:
https://pypi.org/project/svgpathtools/
I'll write this in the README.

I use them to find if path contain another paths. I didn't found this
functionality in existing packages. There are only segments and I didn't
found how to combine then in whole path. This is why I make paths from "d"
property using svgpathtools.
This is not optimal, but for now it works. If someone can optimize the code
it will be great!

Best regards!


On Wed, Dec 5, 2018 at 2:01 AM paulusjacobus <notifications@github.com>
wrote:

> Hi Stefanov,
>
> Thanks for making this available but I do get one error: from svgpathtools
> import *
> the svgpathtools is not defined and the scripts abort from that. Please
> can you update the code?
> Thanks heaps!
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/paulusjacobus/Gerbil/pull/2#issuecomment-444306323>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AASe1rG5j_0ta3_AXvPiT_I_6nc38UzHks5u1wzhgaJpZM4ZBrBv>
> .
>


-- 
Stefan Stefanov

paulusjacobus commented:
That's awesome Stefanov! I will test it out tomorrow morning and release it on the downloads page with instructions when everything works fine. It's a great add on to have!

Thanks, Paul

sstefanov committed:
Some speed optimizations

Removed unnecessary path checks

sstefanov committed:
Removed objbrowser

sstefanov committed:
Added tree sort

Default sorting is tree sort by distance. First cut most inner paths, then raise up to parent object, then get next closest object and cut first it's most inner paths, etc.

sstefanov commented:
I will make new pull request with better sorting.
